### PR TITLE
Add 3d tilesset

### LINKED ISSUE #3448: Mouse over / no object info in LUBIS
We have a inconsistency

https://s.geo.admin.ch/6decadaa7c

on mousoever  and click I get some areas with "hand active / no objetc info" for areas with no data
image

and for some are with no area not
image

moreover some customer report that they  _always_ get no object info even if they click on icons... But I could not reproduce  on chrome or ie

### LINKED ISSUE #3472: Time selector disappears in IE9
IE only

im IE verschwindet das Fenster mit den Jahreszahlen (Zeitreise-Kartenwerke), sobald der Scrollbalken betätigt wird.
Mit Firefox und Chrome funktioniert es problemlos.

imagete

### LINKED ISSUE #3567: 3D: void layer mode
use case: show addresses in 3D
since - as defined- the buildings are on top of the drapped WMS layer, suer can't see the data.
therefore a switch to void layer has to be done
https://mf-geoadmin3.int.bgdi.ch/3dtiles_shadows/index.html?topic=ech&lang=en&bgLayer=voidLayer&layers_visibility=false,true,false,false&layers_timestamp=18641231,,,&lon=8.30863&lat=47.03557&elevation=661&heading=0.498&pitch=-38.105&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege

result:
image

expected result:
iv oid layer in 3d no buildings are shown 

question: does this proposal make sense?

### LINKED ISSUE #3577: Remove KML hacks
[Here](https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/map/KmlService.js#L44) and [here](https://github.com/geoadmin/mf-geoadmin3/blob/master/src/components/map/KmlService.js#L154)

A PR in OpneLayers is needed

### LINKED ISSUE #3616: Embed- preview on small screens not available
open https://map.geo.admin.ch/?topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,

reduce width / small desktop 

embed preview
image

expected result:
Previwe button is visible as copy link button

### LINKED ISSUE #3672: tooltip: highlight mouse
only chrome / not IE
open this  wanderland layer

https://map.geo.admin.ch/?topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege,ch.astra.wanderland&layers_visibility=false,false,false,false,true&layers_timestamp=18641231,,,,&X=201968.70&Y=676652.10&zoom=13

on mouseover, mouse transforms to a hand  on clicking

result:
no tooltip

expected result;
no hand transformation or infotootip: no object info

### LINKED ISSUE #3673: [WMS] Proposed WMS -> new WMS VD
Hi everyone!

I Wish you all the best for 2017!
As a New Year gift, Canton Vaud updated its WMS service! 

Old url http://wms.vd.ch/public/services/VD_WMS/Mapserver/Wmsserver?
should be replaced with:
http://wms.vd.ch/public/services/wmsVD/Mapserver/Wmsserver
on:
https://github.com/geoadmin/mf-geoadmin3/blob/master/src/js/ImportWmsController.js#L47

Bonne année 2017 @ tous!

Arnaud

See 
https://www.asitvd.ch/chercher/geoservices/geoservice-de-consultation-wms.html for more details.

and check out https://www.asitvd.ch/chercher/geoservices/fond-de-plan-asit-vd.html for nice background layers.

### LINKED ISSUE #3675: 3d Icon
hi we need
3d instead of 2.512236d

image

active and not active button

### LINKED ISSUE #3685: WMS 3D not loaded URI
https://mf-geoadmin3.int.bgdi.ch/3dtiles/index.html?topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=WMS%7C%7Cswisscom_leitungen%7C%7Chttp:%2F%2Fservice.lisag.ch%2Fwms%7C%7Cleitungen:swisscom_leitungen%7C%7C1.3.0&lon=8.65239&lat=46.85695&elevation=637&heading=360.000&pitch=-40.000

### PR DESCRIPTION
**Pull request by oterral**
[Test](https://mf-geoadmin3.int.bgdi.ch/3dtiles/index.html)

Missing:

 - [x] 3d button
 - [x] layer without footprint (will be deploy on wednesday)
 - [x] need to update ol3-cesium for https://github.com/openlayers/ol3-cesium/pull/433
 - [x] Open 3d with voidLayer bg fails

To test:

 - Context popup
 - Geolocation (fly to animation)
 - Import KML 
 - Draw
 - Search (fly to animation)
 - 3d buildings

Fix #3567 
Fix #3680
Fix #3577
Fix #3448
Fix #3616 
Fix #3672 
Fix #3673 
Fix #3472
Fix #3685 
Fix #3675

### CODE REVIEW

**Comment by loicgasser** [comment]
I will take the time to review/test it on Thursday and Friday.

File: src/components/map/MapService.js [hunk 1]
```diff
@@ -548,6 +561,38 @@ goog.require('ga_urlutils_service');
...
+                  response.data[tileset3dId] = {
+                    type: 'tileset3d',
+                    serverLayerName: tileset3dId,
+                    timestamps: [tilesetTs[idx]],
+                    attribution: 'swisstopo',
+                    attributionUrl: 'https://www.swisstopo.admin.ch/' + lang +
+                        '/home.html'
+                  };
+                }
+              });
```
Comment (loicgasser) [inline src/components/map/MapService.js]: Is there a plan to put these configs in the service?

**Comment by loicgasser** [comment]
Also I would really deactivate the select by rectangle as it messes up with the default nav and is kind of useless in 3d. ping @davidoesch

File: src/components/map/MapService.js (hunk 1 above)
Comment (oterral) [inline src/components/map/MapService.js]: As far as I know no. But we should before the official release of 3d.

There is also others hardcoded layers thing in this file we could move to the service or db

**Comment by davidoesch** [comment]
@loicgasser : the select by rectangle in 3d you can deactivate

**Comment by davidoesch** [comment]
@oterral https://github.com/geoadmin/mf-geoadmin3/issues/3472 dos not work..still disappearing on IE here

**Comment by oterral** [comment]
@davidoesch I forget to deploy the branch, can you retry now please

**Comment by davidoesch** [comment]
Still doesnt work

**Comment by oterral** [comment]
hmmm sometimes it works,  sometimes not.

File: src/js/GaCesium.js [hunk 2]
```diff
@@ -115,7 +125,61 @@ var GaCesium = function(map, gaPermalink, gaLayers, gaGlobalOptions,
...
+            },
+            show: '${Height} > 0',
+            meta: {
+              description:'"Building id ${id} has height ${Height}."'
+            }
+          });*/
+        }
+        if (tileset) {
+          scene.primitives.add(tileset);
+          primitives.push(tileset);
```
Comment (oterral) [inline src/js/GaCesium.js]: that was my first idea but we don't know what could be inside `scene.primitives` in the future and how differentiate each tileset to remove to the others so, to be sure, I use another array.

**Comment by oterral** [comment]
@davidoesch this time I thnik it's good that changes a bit the behavior (popup stay opens when you move the map) .

Don't forget to clear cache

File: src/components/map/MapService.js (hunk 1 above)
Comment (loicgasser) [inline src/components/map/MapService.js]: That might be a tiny short right now but yes all at once.

**Comment by davidoesch** [comment]
Hi, on mobile this branch does not load the buildings. 3demo does

**Comment by loicgasser** [comment]
test link dead? https://mf-geoadmin3.int.bgdi.ch/3dtiles/index.html

**Comment by oterral** [comment]
was deploying, for the 3d button

**Comment by oterral** [comment]
jenkins test

**Comment by loicgasser** [comment]
:champagne:

**Pull request merged**